### PR TITLE
upgrade google-java-storage

### PR DESCRIPTION
**Pull request by singhravidutt**
Ignore some of the test where Aunutorized checks were made as theose are not supported in gRPC-DirectPath flow.

### CODE REVIEW

**Review by abmodi — APPROVED** [review]

File: tools/run_integration_tests.sh [hunk 1]
```diff
@@ -75,5 +75,6 @@ fi
 export GCS_TEST_PRIVATE_KEYFILE
 export HDFS_ROOT=file:///tmp
 export RUN_INTEGRATION_TESTS=true
+export GOOGLE_CLOUD_ENABLE_DIRECT_PATH_XDS=true
```
Comment (arunkumarchacko) [inline tools/run_integration_tests.sh:78]: Please add a comment explaining this.

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemHTTPClientIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.google.cloud.hadoop.fs.gcs;
```
Comment (arunkumarchacko) [inline gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemHTTPClientIntegrationTest.java:1]: file header. Other files as well.

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GoogleHadoopFileSystemJavaStorageClientIntegrationTest
+    extends GoogleHadoopFileSystemIntegrationTest {
+
+  @Before
+  public void before() throws Exception {
+    storageClientType = ClientType.STORAGE_CLIENT;
```
Comment (arunkumarchacko) [inline gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java:16]: having a mutable field is an anti pattern. A template method might be more appropriate here.

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageFileSystemOptions.ClientType;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GoogleHadoopFileSystemJavaStorageClientIntegrationTest
```
Comment (arunkumarchacko) [inline gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java:11]: Please open a tracking bug to fix the test cases.

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemJavaStorageClientTest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    extends GoogleCloudStorageFileSystemTestBase {
+
+  @Before
+  public void before() throws Exception {
+    storageClientType = ClientType.STORAGE_CLIENT;
+    super.before();
+  }
+
+  @Override
+  @Ignore("DirectPath is not supported with null credentials")
```
Comment (arunkumarchacko) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemJavaStorageClientTest.java:22]: We can have CFE path also for Javastorage client, correct?

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java (hunk 4 above)
Comment (singhravidutt) [reply gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java:28]: added

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemJavaStorageClientTest.java (hunk 5 above)
Comment (singhravidutt) [reply gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemJavaStorageClientTest.java:38]: No. JavaStorage is used specifically for gRPC over directPath and we don't have usecase of gRPC over CFE.

File: tools/run_integration_tests.sh (hunk 1 above)
Comment (singhravidutt) [reply tools/run_integration_tests.sh:79]: done.

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java (hunk 3 above)
Comment (singhravidutt) [reply gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java:33]: It's a temporary fix to have a way to override storageClient. We didn't need it and will mostly clear it up once we have a fix for it.

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemHTTPClientIntegrationTest.java (hunk 2 above)
Comment (singhravidutt) [reply gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemHTTPClientIntegrationTest.java:17]: added

**Pull request merged**